src/autocomplete/EmojiProvider.tsx

```diff
@@ -38,21 +37,26 @@ const EMOJI_REGEX = new RegExp('(' + EMOTICON_REGEX.source + '|(?:^|\\s):[+-\\w]
 
 interface IEmojiShort {
     emoji: IEmoji;
-    shortname: string;
+    shortcode: string;
+    altShortcodes: string[];
     _orderBy: number;
 }
 
-const EMOJI_SHORTNAMES: IEmojiShort[] = EMOJI.sort((a, b) => {
+const EMOJI_SHORTCODES: IEmojiShort[] = EMOJI.sort((a, b) => {
     if (a.group === b.group) {
         return a.order - b.order;
     }
     return a.group - b.group;
-}).map((emoji, index) => ({
-    emoji,
-    shortname: `:${emoji.shortcodes[0]}:`,
-    // Include the index so that we can preserve the original order
-    _orderBy: index,
-}));
+}).map((emoji, index) => {
+    const [shortcode, ...altShortcodes] = getShortcodes(emoji);
+    return {
+        emoji,
+        shortcode: shortcode ? `:${shortcode}:` : undefined,
+        altShortcodes: altShortcodes.map(s => `:${s}:`),
```

Comment:
lets just use an array here, have a comment saying `// we consider shortcodes[0] the canonical one`

Reply:
if we zip the data we don't need to pluck these out, they'll be in `emoji.shortcodes`